src/components/Calculator.js

```diff
@@ -1,41 +1,61 @@
 /* eslint-disable */
 
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.mathMagic = this.mathMagic.bind(this);
+  }
+
+  mathMagic(e) {
+    this.setState(calculate(this.state, e.target.textContent));
+  }
+
   render() {
+    const { total, next, operation } = this.state;
     return (
-      <table className="container">
-        <tr className="screen">0</tr>
-        <tr className="row">
-          <td className="button">AC</td>
-          <td className="button">+/-</td>
-          <td className="button">%</td>
-          <td className="button col-4">&#xf7;</td>
-        </tr>
-        <tr className="row">
-          <td className="button">7</td>
-          <td className="button">8</td>
-          <td className="button">9</td>
-          <td className="button col-4">x</td>
-        </tr>
-        <tr className="row">
-          <td className="button">4</td>
-          <td className="button">5</td>
-          <td className="button">6</td>
-          <td className="button col-4">-</td>
-        </tr>
-        <tr className="row">
-          <td className="button">1</td>
-          <td className="button">2</td>
-          <td className="button">3</td>
-          <td className="button col-4">+</td>
-        </tr>
-        <tr className="row">
-          <td className="button span-2">0</td>
-          <td className="button">.</td>
-          <td className="button col-4">=</td>
-        </tr>
+      <table>
+        <tbody>
+          <tr>
+            <td className="screen" colspan="4">{total}{operation}{next}</td>
+          </tr>
+          <tr>
+            <td onClick={this.mathMagic}>AC</td>
+            <td onClick={this.mathMagic}>+/-</td>
+            <td onClick={this.mathMagic}>%</td>
+            <td className="col-4" onClick={this.mathMagic}>&#xf7;</td>
+          </tr>
+          <tr>
+            <td onClick={this.mathMagic}>7</td>
+            <td onClick={this.mathMagic}>8</td>
+            <td onClick={this.mathMagic}>9</td>
+            <td className="col-4" onClick={this.mathMagic}>x</td>
+          </tr>
+          <tr>
+            <td onClick={this.mathMagic}>4</td>
+            <td onClick={this.mathMagic}>5</td>
+            <td onClick={this.mathMagic}>6</td>
+            <td className="col-4">-</td>
```

Comment:
- [ ] Please add the functionality to the subtraction `-` button because clicking on it doesn't do anything.